### PR TITLE
Fix Boolean Nulls/Refactor Time Adapters/Adapter Tests

### PR DESCRIPTION
**Pull request by SentryMan**
As the title says.

- Fixes null handling with boolean values
- fix Min/Max adapters casting issue (can't cast Object to long if it's an int)
- refactor FuturePastAdapter to be simple and use a referenceClock for tolerance (somewhat similar algo to how hibernate does it)
- no longer uses reflection to check array length

### CODE REVIEW

**Comment by SentryMan** [comment]
hmm, it seems we will need a way to set the tolerance for the time adapters

**Comment by SentryMan** [comment]
Yeah, I'm good with this

**Comment by SentryMan** [comment]
oh wait I forgot the number ones

File: validator/src/main/java/io/avaje/validation/core/adapters/BasicAdapters.java [hunk 1]
```diff
@@ -54,7 +50,11 @@ private static final class PatternAdapter implements ValidationAdapter<CharSeque
 
     @Override
     public boolean validate(CharSequence value, ValidationRequest req, String propertyName) {
-      if (value == null || pattern.test(value.toString())) {
+      if (value == null) {
+        return true;
+      }
+
+      if (pattern.test(value.toString())) {
```
Comment (rbygrave) [inline validator/src/main/java/io/avaje/validation/core/adapters/BasicAdapters.java:57]: This looks wrong / inverted boolean ?  Kind of expecting a test to fail ?  Hmm.

File: validator/src/main/java/io/avaje/validation/core/adapters/BasicAdapters.java [hunk 2]
```diff
@@ -220,4 +224,27 @@ public boolean validate(Object value, ValidationRequest req, String propertyName
       return true;
     }
   }
+
+  private static int arrayLength(Object array) {
```
Comment (rbygrave) [inline validator/src/main/java/io/avaje/validation/core/adapters/BasicAdapters.java:228]: I'm not sure we desire this over `Array.getLength(value);` ?

**Review by rbygrave — APPROVED** [review]

File: validator/src/main/java/io/avaje/validation/core/adapters/BasicAdapters.java (hunk 1 above)
Comment (rbygrave) [reply validator/src/main/java/io/avaje/validation/core/adapters/BasicAdapters.java:57]: Ok, there is the negate() there ... not sure how the tests previously passed?

**Pull request merged**